src/main/java/com/rackspacecloud/metrics/ingestionservice/influxdb/providers/NotcreatedDBInfluxDB.java

```diff
@@ -0,0 +1,230 @@
+
+        try {
+            if (!influxDB.databaseExists("db_1")) {
+                //  influxDB.createDatabase(database);
+                String createRetPolicyString =
+                        String.format("CREATE RETENTION POLICY \"non_existing_rp\" " +
+                                "ON \"%s\" DURATION 5d REPLICATION 1 DEFAULT", "db_1");
+                // Get Column values
+                // Query query = new Query("SHOW MEASUREMENTS", database);
+                // }else {
+                //   System.out.println("DB already exists");
+            }
+        } catch (Exception e) {
+            // The database may set up dynamic proxy, and does not support creating database.
+            e.printStackTrace();
+        }
+        // Read list of DBs in the InfluxDb
+        Query dbquery = new Query("SHOW DATABASES", null);
+        QueryResult queryResult = influxDB.query(dbquery);
+        List<QueryResult.Result> dbresults = queryResult.getResults();
+        List<List<Object>> finaldbresults = dbresults.get(0).getSeries().get(0).getValues();
+        int bkdbcount = 0;
+        int bktblecount = 0;
+        for (int i = 0; i < finaldbresults.size(); i++) {
+            List<Object> dbnames = finaldbresults.get(i);
+            for (Object dbname : dbnames) {
+                //Query query = new Query("SHOW MEASUREMENTS", dbname.toString());
+                Query query = new Query("SHOW MEASUREMENTS", "db_0");
+                List<QueryResult.Result> results = influxDB.query(query).getResults();
+                List<List<Object>> rawMetricNames = results.get(0).getSeries().get(0).getValues();
+
+                bktblecount = 0;
+                for (Object bktable : rawMetricNames) {
+                    // System.out.println("Total number of databases in DB_0  ----" + rawMetricNames);
+                    if (rawMetricNames.isEmpty() || rawMetricNames == null) {
+                        System.out.println("Tables not available for ----" + dbname);
+                    } else {
+                        // System.out.println("Tables available in  ----" + rawMetricNames);
+                    }
+                    bktblecount = bktblecount+1;
+                    //  }
+                }
+            }
+            bkdbcount = i + 1;
+        }
+        //System.out.println("Total DB count in backup db---" + bkdbcount);
+        //System.out.println("Total table count in backup db---" + bktblecount);
+
+        influxDB.createDatabase("db_1");
+        //  System.out.println("Restore database has been created");
+
+        // Copy all measurements from db_0 to Db_1 restore database.
+        QueryResult copyDb = influxDB.query(new Query("SELECT * INTO db_1.autogen.:MEASUREMENT FROM db_0.rp_5d./.*/ GROUP BY *", "db_0"));
+        // System.out.println("Copy of all databases, measurements and datas has been restored");
+
+        // Compare Backup and the Restore DB
```

Comment:
From the comments and from what I can tell of the `SELECT * INTO ...` this copies db_0 into db_1 and then compares the two? Isn't that always going to pass?

Reply:
@hravi76 db_0 is not a backup db. Jira description also states:"Ingestion service collects and caches incoming data in-memory. After a particular amount of time, that data is dumped into a file into a GCP bucket (one of the backup buckets based on Application.yml configuration)." Then database should be restored from a backup file. 

Please follow tests instruction from here https://github.com/racker/ceres-ingestion-service/blob/master/disaster_recovery.md